### PR TITLE
New Loan Calculator (Interactive)

### PR DESCRIPTION
**Pull request by pjhampton**
## Description of new Instant Answer, or changes

A new interactive loan calculator.

screen shot 2017-08-04 at 11 58 31 pm
screen shot 2017-08-04 at 11 58 07 pm

## Todo

- [ ] Deprecate [Loan.pm](https://duck.co/ia/loan)
- [ ] Port triggering
- [ ] Create suitable mobile interface
- [ ] Refactor

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/loan_calculator

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by pjhampton** [comment]
Anybody want to review this? @duckduckgo/duckduckhack-contributors

**Comment by adityatandon007** [comment]
@pjhampton looks great but in my browser it doesn't look the same as shown in your browser.
This is how it looks:
screenshot from 2017-08-05 14-41-24

**Dark theme:**
screenshot from 2017-08-05 14-03-11
In dark mode I wasn't able to see the text that switch between the two.

Browser: Firefox
Result: From duckpan server 

Is it something because of firefox ?

File: share/goodie/loan_calculator/loan_calculator.css [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+.zci--loan_calculator .loan_bar--info p {
+    font-size: 1.5em
+}
+
+/*
+ * Selector styles
+ */
+
+.zci--loan_calculator .loan_selector {
+    background: #ebebeb;
```
Comment (mintsoft) [inline share/goodie/loan_calculator/loan_calculator.css]: @adityatandon007 I think this is applying even in dark

Comment (pjhampton) [inline share/goodie/loan_calculator/loan_calculator.css]: Weird. If you look at [line 50](https://github.com/duckduckgo/zeroclickinfo-goodies/pull/4403/files#diff-11df15b5197643bb0eea1342896d8eceR50) I try force it in. Will fix soon 👍

**Comment by adityatandon007** [comment]
Yeah I also had a look at it. Its Css looks fine. But the dark mode rule
isn't applying. Looks weird

File: share/goodie/loan_calculator/loan_calculator.css [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    border-color: #c8c8c8;
+    margin: 0.3em 0; 
+    padding: 0.5em 0.3em;
+}
+
+.zci--loan_calculator .dark-bg .loan_selector {
```
Comment (mintsoft) [inline share/goodie/loan_calculator/loan_calculator.css]: is this selector the right way around, I've not checked but I thought it was `.dark-bg .zci--  `

File: share/goodie/loan_calculator/loan_calculator.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
+DDH.loan_calculator = DDH.loan_calculator || {};
+
+(function(DDH) {
+    "use strict";
+
+    var init = false;
```
Comment (mintsoft) [inline share/goodie/loan_calculator/loan_calculator.js]: might as well collapse this into one `var`

File: share/goodie/loan_calculator/loan_calculator.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        $loan_selectors = $local_dom.find(".loan_selector--radio");
+    }
+
+    function registerBehaviours() {
+        $inputs.bind("keyup click change mousewheel", function(_e) {
+            calculateLoanInfo();
+        });
+
+        $loan_selectors.click(function(_e) {
+            if($('#loan_selector--radio-monthly').is(':checked')) {
```
Comment (mintsoft) [inline share/goodie/loan_calculator/loan_calculator.js]: This should probably be under `.zci--loan_calculator`

File: share/goodie/loan_calculator/loan_calculator.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+            repayable = 0;
+            total_cost = 0;
+        }
+        
+        $repayable_output.text(DDG.commifyNumber(repayable));
+        $total_cost_output.text(DDG.commifyNumber(total_cost));
+
+    }
+
+   function calculateAmountBudget(principal, months, rate) {
```
Comment (mintsoft) [inline share/goodie/loan_calculator/loan_calculator.js]: Is principal the right name? I'm used to 'advance' to mean the loan amount, although that might be motor-finance specific

File: share/goodie/loan_calculator/loan_calculator.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+        var payment = 0;
+        var interest = rate / 1200;
+
+        if (principal > 0 && months > 0) {
+            if (rate != 0) {
+                payment = principal / (interest / (1 - Math.pow((1 + interest), -(months))));
```
Comment (mintsoft) [inline share/goodie/loan_calculator/loan_calculator.js]: personally I'd prefer `-1 * months`

File: share/goodie/loan_calculator/loan_calculator.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            } else {
+                payment = principal * months; // 0% interest?
```
Comment (mintsoft) [inline share/goodie/loan_calculator/loan_calculator.js]: Shouldn't this be `principal / months` ?

**Comment by pjhampton** [comment]
Thanks @mintsoft. It's unclear if this will make it to production so I've placed it on hold for now. If it gets the go ahead I will finish it off

**Comment by pjhampton** [comment]
Killing for now, but keeping the branch 👍

**Pull request closed without merging**